### DISCUSSION TITLE
Why does it take ~ 45s to get the first message?

### DISCUSSION THREAD

**Discussion by persona94** [confluentinc/confluent-kafka-python#1766] [Q&A]
I set up the consumer first

```
def setup_consumer(topic_list):
        try:
            consumer = Consumer(
                    <options for broker>,
                    'auto.offset.reset':'earliest',
                    'fetch.min.bytes': '200000',
                }
            )

            consumer.subscribe(topic_list)
        except Exception as e:
            app.logger.error(
                f'Exception during creation of active calls consumer - {e}'
            )
            return

start_consumer(consumer)
```

The consumer code looks like this

```
while True:
            message = consumer.poll(0.1)
            if message is None:
                print('Waiting for message')
                gevent.sleep(0.1)
                continue
            elif message.error():
                print(f'Error in loop {message.error()}')
                break
            else:
              print('Message received')
              ....
```

When I start the consumer it takes at least 45s to see the print "Message received"

**Comment by Aragonski97** [comment]
Do you often restart your code? If so, you might be running into Consumer rebalance, which takes some time to perform. Essentially, if you were connected to the cluster and then suddenly disconnected and reconnected again, the connection between consumer and broker was cut loose, so the broker has to perform consumer rebalance, which means that he has to assign partitions to the consumers left in the consumer group (if any). This usually takes about a minute, depending on your configuration.
